crates/ruff/src/ast/helpers.rs

```diff
@@ -940,7 +940,7 @@ pub fn identifier_range(stmt: &Stmt, locator: &Locator) -> Range {
             | StmtKind::FunctionDef { .. }
             | StmtKind::AsyncFunctionDef { .. }
     ) {
-        let contents = locator.slice(Range::from_located(stmt));
+        let contents = locator.slice(stmt.into());
```

Comment:
Nit: Writing `into` everywhere is a bit cumbersome. You can improve the ergonomics by changing `locator.slice` to 

```rust
    pub fn slice<R: Into<Range>>(&self, range: R) -> &'a str {
        let index = self.get_or_init_index();
		let range = range.into();
        let start = truncate(range.location, index, self.contents);
        let end = truncate(range.end_location, index, self.contents);
        &self.contents[start..end]
    }
```

Reply:
Oh that's nice!

Reply:
What's your opinion on this vs. requiring callers to call `Range::from`, as below?

Reply:
Ha! You caught me.

I don't have a very consistent explanation for it. My thinking is that using the `Into<Range>` "overload" of `slice` pushes the logic and complexity into the `slice` function. The reader of the caller doesn't need to be concerned with the conversation happening. The API of `slice` guarantees that the conversation is OK for all types implementing `Into<Range>`. For as far as readers are concerned, `locator.slice(stmt)` passes a range to `slice`.